src/ralph/models/selector.py

```diff
@@ -0,0 +1,141 @@
+"""Model selector definition"""
+
+import operator
+from collections import Counter
+from dataclasses import dataclass
+from importlib import import_module
+from itertools import chain
+from typing import Any, Callable, Union
+
+from pydantic import BaseModel
+
+from ralph.utils import get_dict_value_from_path
+
+
+@dataclass(frozen=True)
+class ModelField:
+    """Represents a model field."""
+
+    path: tuple[str]
+
+    def __init__(self, path: str):
+        object.__setattr__(self, "path", tuple(path.split(">")))
+
+
+@dataclass(frozen=True)
+class Rule:
+    """Represents a rule used for selection."""
+
+    field: ModelField
+    value: Union[ModelField, Any]
+    operation: Callable[[Any, Any], bool] = operator.eq
+
+    @staticmethod
+    def get_field_value(field, event, known_fields):
+        """Retrieves the value of a rule field.
+
+        Args:
+            field (ModelField or Any): The rule field to retrieve its value.
+            known_fields (dict): A mapping of already retrieved fields of the event.
+        """
+
+        if not isinstance(field, ModelField):
+            return field
+        if field.path in known_fields:
+            return known_fields[field.path]
+        value = get_dict_value_from_path(event, field.path)
+        known_fields[field.path] = value
+        return value
```

Comment:
Good point) Thanks, yes indeed this is confusing for the get_field_value method - removing it.

For the known_fields cache - it's true that for now there is no added value, it might be useful (later) when we will have a lot of rules checking for nested values - but this can be done later if needed - removing it too)